File: server/vue/src/api/control/process_control.js
import request from '@/utils/request'

// 添加单个进程
export function addSingleProcess(data) {
  return request({
    url: '/process/add_single',
    method: 'post',
    data: data
  })
}

// 批量添加进程
export function addBatchProcess(data) {
  return request({
    url: '/process/add_batch',
    method: 'post',
    data: data
  })
}

// 删除单个规则
export function deleteSingleProcess(data) {
  return request({
    url: '/process/delete_single',
    method: 'post',
    data: data
  })
}

// 批量删除规则
export function deleteBatchProcess(data) {
  return request({
    url: '/process/delete_batch',
    method: 'post',
    data: data
  })
}

// 启用/禁用规则
export function toggleProcessStatus(data) {
  return request({
    url: '/process/toggle',
    method: 'post',
    data: data
  })
}

// 获取进程规则列表
export function getProcessList() {
  return request({
    url: '/process/list',
    method: 'get'
  })
}

File: server/fastapi/routers/control/process_control.py
from fastapi import APIRouter, Depends
from model.control.process_model import *
from service.control.process_service import *
from utils.auth import get_current_user
from utils.check_perm import check_permission
from utils.log.log_context import log_context_dependency

router = APIRouter()

# 添加单个
@router.post("/add_single")
async def add_single(req: AddSingleProcessRequest, _=Depends(log_context_dependency),
    _p=Depends(check_permission("process_rule:add"))):
    return add_single_process(req.process_name, status=req.status)

# 批量添加
@router.post("/add_batch")
async def add_batch(req: AddProcessRequest, _=Depends(log_context_dependency),
    _p=Depends(check_permission("process_rule:add"))):
    return add_batch_process(req.process_list, status=req.status)

# 删除单个
@router.post("/delete_single")
async def delete_single(req: DeleteSingleProcessRequest, _=Depends(log_context_dependency),
    _p=Depends(check_permission("process_rule:delete"))):
    return delete_single_process(req.id)

# 批量删除
@router.post("/delete_batch")
async def delete_batch(req: DeleteProcessRequest, _=Depends(log_context_dependency),
    _p=Depends(check_permission("process_rule:delete"))):
    return delete_batch_process(req.ids)

# 启用/禁用
@router.post("/toggle")
async def toggle_status(req: ToggleProcessStatusRequest, _=Depends(log_context_dependency),
    _p=Depends(check_permission("process_rule:toggle"))):
    return toggle_process_status(req.id, req.status)

# 获取全部规则列表
@router.get("/list")
async def list_rules(_=Depends(log_context_dependency),
    _p=Depends(check_permission("process_rule:list"))):
    return get_process_list()
